### PR TITLE
Set up a "Hello world" project with Front End

### PR DESCRIPTION
**Pull request by AlselwiAisha**
### Project requirements

- Create a new Rails API app called 'hello-rails-back-end'.
- Initialize your project with Git.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and rails route).
- Create a new React app called 'hello-react-front-end'
- Initialize your project with Git.
- Create the App component with react-router.
- Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action and a reducer that will connect to you API endpont to get the random greeting.
- Display the random greeting in your Greeting component.
- In the README file for the Rails API, add a link to your React app.
- Also, in the README file for the React app, add a link to your Rails API.
- When submitting for a review - use a link to the pull request in your Rails API.
- In the repo for the frontend part - include a PR with changes.

[Here is Back End](https://github.com/AlselwiAisha/hello-rails-back-end/pull/1)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
+        <Routes>
+          <Route path="/" exact component={Greeting} />
+        </Routes>
       </header>
     </div>
   );
```
Comment (topeogunleye) [inline src/App.js:14]: Hi there,

I've noticed that you've made some changes to the `App.js` file to include Redux and React Router. These are great additions, but I have a few suggestions to improve the code:

1. **Provider Usage**: The `Provider` component from `react-redux` is usually should be added, and the store should be provided from the redux.

3. **BrowserRouter**: You need to add a `BrowserRouter` component, and you haven't imported it. Make sure to import it from `react-router-dom` at the top of your file, and use it or else you could have error with react routers.

Here's how I would suggest modifying your code:

```javascript
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import './App.css';
import Greeting from './Components/Greeting';
import store from './redux/store';

function App() {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <div className="App">
          <header className="App-header">
            <Routes>
              <Route path="/" exact component={Greeting} />
            </Routes>
          </header>
        </div>
      </BrowserRouter>
    </Provider>
  );
}

export default App;

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @AlselwiAisha, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
> * Good commit messages. :heavy_check_mark:
1. **Linking Backend and Frontend:**
   - Confirm that the link to the backend is present in the README of the frontend. This bidirectional linking ensures clarity and accessibility between the frontend and backend components. ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -1,10 +1,106 @@
...
+<!-- LIVE DEMO
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p> -->
+
+<!-- GETTING STARTED -->
+## 🔗 Link To Hello Rails Back-End <a name="link-to-hello-rails-back-end"></a>
+
+Here is the link to the [Hello Rails Back-End](https://github.com/AlselwiAisha/hello-rails-back-end/pull/1)repository.
```
Comment (shahadat3669) [inline README.md]: 👍 Great job on the progress made! To further align with project requirements, consider the following improvements:

- [ ] [Optional] **Link to Backend Repository:** As per project specifications, it's expected to provide a link to the backend repository in your project's README. Currently, the link redirects to the pull request, which doesn't fulfill this requirement. Please update the link to point directly to your backend repository.
image

- [ ] [Optional] Additionally, it is requested that you submit your pull request for the `rails api` in the dashboard when you are requesting a review. For your next project, I recommend that you follow the documentation carefully and ensure that you use the proper link to request a review.
image

By making this adjustment, your project documentation will adhere to the project requirements more closely. Keep up the good work! 🚀👏

File: src/redux/store.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
```
Comment (shahadat3669) [inline src/redux/store.js]: - [ ] Great job so far! I've noticed that your `store` configuration doesn't match the [`redux-thunk`](https://github.com/reduxjs/redux-thunk?tab=readme-ov-file#manual-setup) documentation. I recommend taking a closer look at the documentation and configuring your app accordingly. 
image
- [ ] Additionally, I suggest checking your project locally and addressing any new errors that may arise.

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @AlselwiAisha, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ Worked on previous reviewer comments.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @AlselwiAisha, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Altered the `redux` configuration ✅ 

## Required Changes ♻️

- [ ] Great work, you did a great job, before reading further, please keep in mind you need to carefully read my suggestion on both the repository's PR. Your last code reviewer suggested you implement the redux as per the standards and requirements, but it seems like while making changes to your `front-end` application. You messed up with the configuration, after carefully reviewing your project. I noticed that your redux code is not completely and not wired up properly, which results in not getting any random greeting either on `UI` or `console.log`. Kindly fix the issue and make sure that your application is working as expected and without any problems. Kudos for implementing the backend very well and fixing the cors issue requested by your previous code reviewer.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @AlselwiAisha, 

## Highlights!
- Good code organization and working FE app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**